### PR TITLE
add support for 2ch.hk

### PR DESCRIPTION
**Pull request by anselmov**
Hello. I did some work on that russian board, it allows view & creation of threads and posts.
Login and features which I don't use are not supported, still #542 may be happy with this :)

### CODE REVIEW

**Comment by Floens** [comment]
Looks good. I will verify and merge it when I have the time.

**Comment by MEWop** [comment]
ooohh yeaaah

**Comment by MEWop** [comment]
so long we wait :(

File: Clover/app/src/main/java/org/floens/chan/core/site/sites/dvach/Dvach.java [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+                    call.parameter("post", "New Reply");
+                } else {
+                    call.parameter("post", "New Thread");
+                    call.parameter("page", "1");
+                }
+            }
+
+            @Override
+            public boolean requirePrepare() {
+                // We don't need to check the antispam fields for 8chan.
```
Comment (Floens) [inline Clover/app/src/main/java/org/floens/chan/core/site/sites/dvach/Dvach.java]: Outdated comment?

File: Clover/app/src/main/java/org/floens/chan/core/site/sites/dvach/DvachApi.java [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+                case "files_count":
+                    builder.images(reader.nextInt());
+                    break;
+                case "lasthit":
+                    builder.lastModified(reader.nextLong());
+                    break;
+                case "num":
+                    String num = reader.nextString();
+                    builder.id(Integer.parseInt(num));
+                    builder.posterId(num);
```
Comment (Floens) [inline Clover/app/src/main/java/org/floens/chan/core/site/sites/dvach/DvachApi.java]: `posterId` is used in 4chan for the unique id assigned to unique ip addresses.
The way it is used now causes all posts to have "ID: nnn" on them.

File: Clover/app/src/main/java/org/floens/chan/core/site/sites/dvach/DvachApi.java [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+
+        List<PostImage> files = new ArrayList<>();
+
+        reader.beginObject();
+        while (reader.hasNext()) {
+            String key = reader.nextName();
+
+            switch (key) {
+                case "no":
+                    builder.id(Integer.parseInt(reader.nextString()));
```
Comment (Floens) [inline Clover/app/src/main/java/org/floens/chan/core/site/sites/dvach/DvachApi.java]: I see builder.id being used in both "no" and "num". Why is that?

**Comment by MEWop** [comment]
> Hello. I did some work on that russian board, it allows view & creation of threads and posts.
> Login and features which I don't use are not supported, still #542 may be happy with this :)

Cmon man we need some help

File: Clover/app/src/main/java/org/floens/chan/core/site/sites/dvach/Dvach.java (hunk 1 above)
Comment (anselmov) [reply Clover/app/src/main/java/org/floens/chan/core/site/sites/dvach/Dvach.java]: Probably. I don't know what antispam fields are anyway

File: Clover/app/src/main/java/org/floens/chan/core/site/sites/dvach/DvachApi.java (hunk 3 above)
Comment (anselmov) [reply Clover/app/src/main/java/org/floens/chan/core/site/sites/dvach/DvachApi.java]: 2ch api doesn't have "no" fields, it can be removed

**Comment by MEWop** [comment]
:( :( :( :( :( :( :( :( :(

**Comment by leedoyle** [comment]
Hi, guys! Is there a way to see support for kohlchan.net as well? That's the new Krautchan address.

**Comment by Floens** [comment]
@leedoyle it looks like a vichan site, could you create a new issue for it?

**Pull request merged**